programs/ido-program/tests/ido-program.ts

```diff
@@ -0,0 +1,466 @@
+    createMint,
+    getAccount,
+    getOrCreateAssociatedTokenAccount,
+    mintTo,
+    TOKEN_PROGRAM_ID,
+} from "@solana/spl-token";
+import { Program } from "@project-serum/anchor";
+import { assert } from "chai";
+import { IdoProgram } from "../target/types/ido_program";
+
+describe("ido-program", () => {
+    // Configure the client to use the local cluster.
+    const provider = anchor.AnchorProvider.local();
+    anchor.setProvider(provider);
+
+    const program = anchor.workspace.IdoProgram as Program<IdoProgram>;
+
+    let nativeTokenAmount = new anchor.BN(1000000);
+
+    let fiatMint: PublicKey;
+    let redeemableMint: PublicKey;
+    let nativeMint: PublicKey;
+
+    let projectFiat: PublicKey;
+    let projectNative: PublicKey;
+
+    let investorFiat: PublicKey;
+    let investorNative: PublicKey;
+    let investorRedeemable: PublicKey;
+
+    let poolNative: PublicKey;
+    let poolFiat: PublicKey;
+
+    let poolSigner: PublicKey;
+
+    let nowBn: anchor.BN;
+    let startIdoTs: anchor.BN;
+    let endIdoTs: anchor.BN;
+    let withDrawFiatTs: anchor.BN;
+
+    const payer = anchor.web3.Keypair.generate();
+    const mintAuthority = anchor.web3.Keypair.generate();
+
+    const project = anchor.web3.Keypair.generate();
+    const investor = anchor.web3.Keypair.generate();
+
+    let pool = anchor.web3.Keypair.generate();
+
+    it("Can initialize the program state", async () => {
+        const transferSig = await provider.connection.requestAirdrop(
+            payer.publicKey,
+            10000000000
+        );
+
+        const latestBlockHash = await provider.connection.getLatestBlockhash();
+
+        await provider.connection.confirmTransaction({
+            blockhash: latestBlockHash.blockhash,
+            lastValidBlockHeight: latestBlockHash.lastValidBlockHeight,
+            signature: transferSig,
+        });
+
+        const tx = new Transaction();
+
+        tx.add(
+            SystemProgram.transfer({
+                fromPubkey: payer.publicKey,
+                toPubkey: mintAuthority.publicKey,
+                lamports: 2000000000,
+            }),
+            SystemProgram.transfer({
+                fromPubkey: payer.publicKey,
+                toPubkey: project.publicKey,
+                lamports: 2000000000,
+            }),
+            SystemProgram.transfer({
+                fromPubkey: payer.publicKey,
+                toPubkey: investor.publicKey,
+                lamports: 2000000000,
+            })
+        );
+
+        await provider.sendAndConfirm(tx, [payer]);
+
+        fiatMint = await createMint(
+            provider.connection,
+            payer,
+            mintAuthority.publicKey,
+            undefined,
+            0,
+            undefined,
+            undefined,
+            TOKEN_PROGRAM_ID
+        );
+
+        nativeMint = await createMint(
+            provider.connection,
+            payer,
+            mintAuthority.publicKey,
+            undefined,
```

Comment:
This parameter is type `PublicKey | null`
```suggestion
            null,
```